### PR TITLE
Migrate remainder of THCDeviceUtils.cuh to ATen

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65492
* #65491
* __->__ #65472
* #65471
* #65470

Differential Revision: [D31148181](https://our.internmc.facebook.com/intern/diff/D31148181)
BC-breaking note:
This PR remove THCeilDiv function and THC/THCDeviceUtils.cuh header, please use at::ceil_div and ATen/ceil_div.h header as appropriate

### CODE REVIEW

File: aten/src/ATen/ceil_div.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include <c10/macros/Macros.h>
+
+namespace at {
+
+/**
+   Computes ceil(a / b)
+*/
+template <typename T, typename=std::enable_if_t<std::is_integral<T>::value>>
+C10_ALWAYS_INLINE C10_HOST_DEVICE T ceil_div(T a, T b) {
```
Comment (ngimel) [inline aten/src/ATen/ceil_div.h:10]: There's already ATenCeilDiv in CUDAApplyUtils.cuh already, can you use it? And add `round_up` to that header.

Comment (peterbell10) [reply aten/src/ATen/ceil_div.h:10]: I've done it the other way around, replacing ATenCeilDiv with `ceil_div`. My reasoning is that this isn't cuda specific, and would be useful to have available for future CPU code. Also, that including all of `CUDAApplyUtils.cuh` just for a small helper function is unnecessary.

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148181).

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148181).

**Comment by ngimel** [comment]
Torchvision unfortunately is using ATenCeilDiv https://github.com/pytorch/vision/blob/7947fc8fb38b1d3a2aca03f22a2e6a3caa63f2a0/torchvision/csrc/ops/cuda/interpolate_aa_kernels.cu#L266, for some reason this wasn't caught by the tests

**Comment by ngimel** [comment]
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148181).

**Comment by ngimel** [comment]
A lot of THCeilDivs internally, I'm fixing them, will land on Monday.

**Pull request closed without merging**